### PR TITLE
Adding Integration Tests - (Part 3)

### PR DESCRIPTION
**Pull request by Chamindu36**
## Purpose
Adding Integration tests to apictl integration test suite

## Goals
Automating Integration tests

### Covered Automated Tests from this PR

**Tests Related to APIs**

1. Delete API with a active subscription (TestDeleteApiWithActiveSubscriptionsSuperTenantUser)
2. Change Life Cycle state of an API as Admin super tenant user (TestChangeLifeCycleStatusOfApiAdminSuperTenantUser)
3. Change Life Cycle state of an API as Devops super tenant user (TestChangeLifeCycleStatusOfApiDevopsSuperTenantUser)
4. Change Life Cycle state of an API as Admin tenant user (TestChangeLifeCycleStatusOfApiAdminTenantUser)
5. Change Life Cycle state of an API as Devops tenant user (TestChangeLifeCycleStatusOfApiDevopsTenantUser)
6. Change Life Cycle state of an API as a user without proper permissions (TestChangeLifeCycleStatusOfApiFailWithAUserWithoutPermissions)
7. Change Life Cycle state of an API with active subscriptions as Admin super tenant user (TestChangeLifeCycleStatusOfApiWithActiveSubscriptionWithAdminSuperTenantUser)
8. Change Life Cycle state of an API with active subscriptions as Devops super tenant user (TestChangeLifeCycleStatusOfApiWithActiveSubscriptionWithDevopsSuperTenantUser)

**Tests Related to API Products**

1. Delete API Product with a active subscription (TestDeleteApiProductWithActiveSubscriptionsSuperTenantUser)

**Tests Related to Keys**

1. Get keys consecutively for an API (TestGetKeysConsecutivelyAdminSuperTenantUser)
2. Get keys consecutively for an API Product (TestGetKeysConsecutivelyForAPIProductAdminSuperTenantUser)

## Test environment
OS - Ubuntu 20.04 LTS
Java - JDK 1.8_252
APIM - 3.2.0 -RC5

### CODE REVIEW

**Review by uvindra — APPROVED** [review]

File: import-export-cli/integration/api_test.go [hunk 1]
```diff
@@ -738,3 +771,239 @@ func TestExportApisWithExportApisCommand(t *testing.T) {
...
+	apiCreator := creator.UserName
+	apiCreatorPassword := creator.Password
+
+	dev := apimClients[0]
+
+	// Add the API to env
+	api := testutils.AddAPI(t, dev, apiCreator, apiCreatorPassword)
+
+	//Change life cycle state of Api from CREATED to PUBLISHED
+	args := &testutils.ApiChangeLifeCycleStatusTestArgs{
```
Comment (isharac) [inline import-export-cli/integration/api_test.go:823]: needs a space before {

File: import-export-cli/integration/api_test.go [hunk 2]
```diff
@@ -738,3 +771,239 @@ func TestExportApisWithExportApisCommand(t *testing.T) {
...
+		APIM:          dev,
+		Api:           api,
+		Action:        "Publish",
+		ExpectedState: "PUBLISHED",
+	}
+
+	testutils.ValidateChangeLifeCycleStatusOfAPI(t, args)
+
+	//Change life cycle state of Api from PUBLISHED to CREATED
+	argsToNextChange := &testutils.ApiChangeLifeCycleStatusTestArgs{
```
Comment (isharac) [inline import-export-cli/integration/api_test.go:834]: needs a space before {

File: import-export-cli/integration/api_test.go [hunk 3]
```diff
@@ -738,3 +771,239 @@ func TestExportApisWithExportApisCommand(t *testing.T) {
...
+	apiCreator := creator.UserName + "@" + TENANT1
+	apiCreatorPassword := creator.Password
+
+	dev := apimClients[0]
+
+	// Add the API to env
+	api := testutils.AddAPI(t, dev, apiCreator, apiCreatorPassword)
+
+	//Change life cycle state of Api from CREATED to PUBLISHED
+	args := &testutils.ApiChangeLifeCycleStatusTestArgs{
```
Comment (isharac) [inline import-export-cli/integration/api_test.go:858]: needs a space before {

File: import-export-cli/integration/api_test.go [hunk 4]
```diff
@@ -738,3 +771,239 @@ func TestExportApisWithExportApisCommand(t *testing.T) {
...
+		APIM:          dev,
+		Api:           api,
+		Action:        "Publish",
+		ExpectedState: "PUBLISHED",
+	}
+
+	testutils.ValidateChangeLifeCycleStatusOfAPI(t, args)
+
+	//Change life cycle state of Api from PUBLISHED to CREATED
+	argsToNextChange := &testutils.ApiChangeLifeCycleStatusTestArgs{
```
Comment (isharac) [inline import-export-cli/integration/api_test.go:869]: needs a space before {

File: import-export-cli/integration/api_test.go [hunk 5]
```diff
@@ -738,3 +771,239 @@ func TestExportApisWithExportApisCommand(t *testing.T) {
...
+
+	apiCreator := creator.UserName + "@" + TENANT1
+	apiCreatorPassword := creator.Password
+
+	dev := apimClients[0]
+	// Add the API to env
+	api := testutils.AddAPI(t, dev, apiCreator, apiCreatorPassword)
+
+	//Change life cycle state of Api from CREATED to PUBLISHED
+	args := &testutils.ApiChangeLifeCycleStatusTestArgs{
```
Comment (isharac) [inline import-export-cli/integration/api_test.go:892]: needs a space before {

File: import-export-cli/integration/api_test.go [hunk 6]
```diff
@@ -738,3 +771,239 @@ func TestExportApisWithExportApisCommand(t *testing.T) {
...
+		APIM:          dev,
+		Api:           api,
+		Action:        "Publish",
+		ExpectedState: "PUBLISHED",
+	}
+
+	testutils.ValidateChangeLifeCycleStatusOfAPI(t, args)
+
+	//Change life cycle state of Api from PUBLISHED to CREATED
+	argsToNextChange := &testutils.ApiChangeLifeCycleStatusTestArgs{
```
Comment (isharac) [inline import-export-cli/integration/api_test.go:903]: needs a space before {

File: import-export-cli/integration/api_test.go [hunk 7]
```diff
@@ -738,3 +771,239 @@ func TestExportApisWithExportApisCommand(t *testing.T) {
...
+
+	apiCreator := creator.UserName + "@" + TENANT1
+	apiCreatorPassword := creator.Password
+
+	dev := apimClients[0]
+	// Add the API to env
+	api := testutils.AddAPI(t, dev, apiCreator, apiCreatorPassword)
+
+	//Change life cycle state of Api from CREATED to PUBLISHED
+	args := &testutils.ApiChangeLifeCycleStatusTestArgs{
```
Comment (isharac) [inline import-export-cli/integration/api_test.go:926]: needs a space before {

File: import-export-cli/integration/api_test.go [hunk 8]
```diff
@@ -738,3 +771,239 @@ func TestExportApisWithExportApisCommand(t *testing.T) {
...
+	apiCreator := creator.UserName
+	apiCreatorPassword := creator.Password
+
+	dev := apimClients[0]
+	// Add the API to env
+	api := testutils.AddAPI(t, dev, apiCreator, apiCreatorPassword)
+
+	testutils.PublishAPI(dev, adminUsername, adminPassword, api.ID)
+
+	args := &testutils.ApiGetKeyTestArgs{
```
Comment (isharac) [inline import-export-cli/integration/api_test.go:950]: needs a space before {

File: import-export-cli/integration/api_test.go [hunk 9]
```diff
@@ -738,3 +771,239 @@ func TestExportApisWithExportApisCommand(t *testing.T) {
...
+		Apim:    dev,
+	}
+
+	//Create an active subscription for Api
+	testutils.ValidateGetKeysWithoutCleanup(t, args)
+
+	base.WaitForIndexing()
+
+	//Change life cycle state of Api from PUBLISHED to CREATED
+	argsToLifeCycleStateChange := &testutils.ApiChangeLifeCycleStatusTestArgs{
```
Comment (isharac) [inline import-export-cli/integration/api_test.go:962]: needs a space before {

File: import-export-cli/integration/api_test.go [hunk 10]
```diff
@@ -738,3 +771,239 @@ func TestExportApisWithExportApisCommand(t *testing.T) {
...
+	apiCreator := creator.UserName
+	apiCreatorPassword := creator.Password
+
+	dev := apimClients[0]
+	// Add the API to env
+	api := testutils.AddAPI(t, dev, apiCreator, apiCreatorPassword)
+
+	testutils.PublishAPI(dev, devopsUsername, devopsPassword, api.ID)
+
+	args := &testutils.ApiGetKeyTestArgs{
```
Comment (isharac) [inline import-export-cli/integration/api_test.go:987]: needs a space before {

File: import-export-cli/integration/api_test.go [hunk 11]
```diff
@@ -738,3 +771,239 @@ func TestExportApisWithExportApisCommand(t *testing.T) {
...
+		Apim:    dev,
+	}
+
+	//Create an active subscription for Api
+	testutils.ValidateGetKeysWithoutCleanup(t, args)
+
+	base.WaitForIndexing()
+
+	//Change life cycle state of Api from PUBLISHED to CREATED
+	argsToLifeCycleStateChange := &testutils.ApiChangeLifeCycleStatusTestArgs{
```
Comment (isharac) [inline import-export-cli/integration/api_test.go:999]: needs a space before {

**Pull request merged**